### ISSUE TITLE
3D Particle shadows not updating

### ISSUE THREAD

**Issue by BlackJitsu** [godotengine/godot#17267]
I set up a 3D scene with the default environment, a player with a camera attached, 3D particles that emit the default spheres, and a bouncing ball. Using the directionalLight node , the shadow for the bouncing ball updates even when out of the frame, but the particle shadows do not. 

https://imgur.com/a/rFZlT

**Comment by eon-s** [comment]
Godot version? and can you post a minimal example to test?

Also, what happens if you increase the aabb of the particles node? because looks like particles stopped updating at all when you got too far.

**Comment by eon-s** [comment]
Tried to reproduce this and seems to be related to the particles visibility AABB being outside of camera range, making the Particles node to stop updating but the shadow of the frozen particles remain there, not sure if this is a bug because it shows the correct particles behavior (and the only thing currently working with the AABB).

I think the way to prevent this is to have a particles AABB big enough to cover the shadow coming from the back of the camera.

![particlesaabb](https://user-images.githubusercontent.com/10215987/37065720-be0a6bfe-2180-11e8-8933-24b3f6b12610.gif)

**Comment by lawnjelly** [comment]
Increasing the AABB is the wrong way to deal with this. The particles should be updated when in view of either the camera or lights (culled against the view frustum).

**Comment by akien-mga** [comment]
Reopening as per @lawnjelly's comment.

**Comment by lawnjelly** [comment]
> Reopening as per @lawnjelly's comment.

Yes, this may actually not be that difficult to fix, presumably there needs to be an animated flag on particles and skinned meshes so that they update in the same way as if you had moved them (I think this sets a dirty flag if I remember right).

**Comment by Calinou** [comment]
I can confirm this on 4.0.alpha11 with both GPUParticles3D and CPUParticles3D. Depending on the type of particles, the shadow updating is different.

GPUParticles3D in yellow (left), CPUParticles3D in blue (right):

https://user-images.githubusercontent.com/180032/178158022-e18406c9-069c-4926-be2c-b71e0bf07ab7.mp4

GPUParticles3D stop updating entirely when the AABB is outside the camera's frustum, but CPUParticles3D's shadow will still update. *However*, low processor mode will not take this into account, so the shadow will only update if the camera is moving or the editor is redrawing for some other reason (e.g. hovering a button).
